bundles/org.openhab.binding.zm/src/main/java/org/openhab/binding/zm/internal/handler/ZmBridgeHandler.java

```diff
@@ -0,0 +1,648 @@
+                updateState(urlChannelId, new StringType(buildStreamUrl(id, type)));
+            } else {
+                updateState(monitorIdChannelId, UnDefType.UNDEF);
+                updateState(urlChannelId, UnDefType.UNDEF);
+            }
+        }
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(MonitorDiscoveryService.class);
+    }
+
+    public void setDiscoveryService(MonitorDiscoveryService discoveryService) {
+        this.discoveryService = discoveryService;
+    }
+
+    public boolean isDiscoveryEnabled() {
+        return discoveryEnabled;
+    }
+
+    public Integer getDefaultAlarmDuration() {
+        return defaultAlarmDuration;
+    }
+
+    public @Nullable Integer getDefaultImageRefreshInterval() {
+        return defaultImageRefreshInterval;
+    }
+
+    public List<Monitor> getSavedMonitors() {
+        return savedMonitors;
+    }
+
+    public Gson getGson() {
+        return GSON;
+    }
+
+    public void setFunction(String id, MonitorFunction function) {
+        if (!zmAuth.isAuthorized()) {
+            return;
+        }
+        logger.debug("Bridge: Setting monitor {} function to {}", id, function);
+        executePost(buildUrl(String.format("/api/monitors/%s.json", id)),
+                String.format("Monitor[Function]=%s", function.toString()));
+    }
+
+    public void setEnabled(String id, OnOffType enabled) {
+        if (!zmAuth.isAuthorized()) {
+            return;
+        }
+        logger.debug("Bridge: Setting monitor {} to {}", id, enabled);
+        executePost(buildUrl(String.format("/api/monitors/%s.json", id)),
+                String.format("Monitor[Enabled]=%s", enabled == OnOffType.ON ? "1" : "0"));
+    }
+
+    public void setAlarmOn(String id) {
+        if (!zmAuth.isAuthorized()) {
+            return;
+        }
+        logger.debug("Bridge: Turning alarm ON for monitor {}", id);
+        setAlarm(buildUrl(String.format("/api/monitors/alarm/id:%s/command:on.json", id)));
+    }
+
+    public void setAlarmOff(String id) {
+        if (!zmAuth.isAuthorized()) {
+            return;
+        }
+        logger.debug("Bridge: Turning alarm OFF for monitor {}", id);
+        setAlarm(buildUrl(String.format("/api/monitors/alarm/id:%s/command:off.json", id)));
+    }
+
+    public @Nullable RawType getImage(String id, @Nullable Integer imageRefreshInterval) {
+        Integer localRefreshInterval = imageRefreshInterval;
+        if (localRefreshInterval == null || localRefreshInterval.intValue() < 1 || !zmAuth.isAuthorized()) {
+            return null;
+        }
+        // Call should timeout just before the refresh interval
+        int timeout = (localRefreshInterval * 1000) - 500;
+        Request request = httpClient.newRequest(buildStreamUrl(id, STREAM_IMAGE));
+        request.method(HttpMethod.GET);
+        request.timeout(timeout, TimeUnit.MILLISECONDS);
```

Comment:
What happens if the user configures a refresh interval of 10min? You could set an upper limit to the timeout.

Reply:
Good idea.